report/3.Theory/LanguageSpecification.tex

```diff
@@ -88,6 +100,22 @@ \subsection{Functions}
 
 The \texttt{<DclValue>} non-terminal refers to both \texttt{<Expr>} (expression) for type declarations and \texttt{<LambdaExpr>} (function parameters and body) for function declarations.
 
+The below listing showcases declarations in HCL using type inference as described above.
+The listing includes both assignment of expressions and assignment of lambda-expressions.
+
+\begin{lstlisting}[language=HCL,caption={Implicit type declarations in HCL.
+},firstnumber=1]
+var x = 3141596
+var y = 2718281
+var z = x > y
+
+var w = x + y
+
+var v = (num v, num u): { v * u}
+
+var h = (txt h)
```

Comment:
what. this line is not valid from the parser.. x) you can't have a partial lambda decleration. also the previous line lacks a return type.

Reply:
Fixed